src/ezdxf/tools/text.py

```diff
@@ -414,8 +416,18 @@ def plain_text(text: str) -> str:
 # - Paragraphs do overflow into the next column if required.
 
 
+@overload
+def fast_plain_mtext(text: str, split: Literal[True]) -> list[str]:
+    ...
+
+
+@overload
+def fast_plain_mtext(text: str, split: Literal[False] = False) -> str:
+    ...
+
+
 # pylint: disable-next=too-many-branches
-def fast_plain_mtext(text: str, split=False) -> Union[list[str], str]:
+def fast_plain_mtext(text: str, split: bool = False) -> Union[list[str], str]:
```

Comment:
I don't add `bool` for such cases, type inference by `False` and `True`, shorter signatures are better, therefore return `list[str]|str` .

Reply:
I'd argue that shorter signatures aren't better when compared against actually accurate signatures.
Type inference from `False` is `Any` in mypy (not mentioning it erroring in strict mode).

Reply:
> therefore return `list[str]|str`
Migrating to PEP 604 is a topic for a different PR.